subsys/bluetooth/mesh/gen_onoff_cli.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <bluetooth/mesh/gen_onoff_cli.h>
+#include "model_utils.h"
+
+static void decode_status(struct net_buf_simple *buf,
+			  struct gen_onoff_status *status)
+{
+	status->present_on_off = net_buf_simple_pull_u8(buf);
+
+	if (buf->len == 2) {
+		status->target_on_off = net_buf_simple_pull_u8(buf);
+		status->remaining_time =
+			model_transition_decode(net_buf_simple_pull_u8(buf));
+	} else {
+		status->target_on_off = status->present_on_off;
+		status->remaining_time = 0;
+	}
+}
+
+static void handle_status(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	if (buf->len != 1 && buf->len != 3) {
+		return;
+	}
+
+	struct gen_onoff_cli *cli = model->user_data;
+	struct gen_onoff_status status;
+
+	decode_status(buf, &status);
+
+	if (model_ack_match(&cli->ack_ctx, GEN_ONOFF_OP_STATUS, ctx)) {
+		struct gen_onoff_status *rsp =
+			(struct gen_onoff_status *)cli->ack_ctx.param;
+
+		*rsp = status;
+		model_ack_rx(&cli->ack_ctx);
+	} else if (cli->status_handler) {
+		cli->status_handler(cli, ctx, &status);
+	}
+}
+
+const struct bt_mesh_model_op _gen_onoff_cli_op[] = {
+	{ GEN_ONOFF_OP_STATUS, 1, handle_status },
+	BT_MESH_MODEL_OP_END,
+};
+
+static int gen_onoff_cli_init(struct bt_mesh_model *model)
+{
+	struct gen_onoff_cli *cli = model->user_data;
+
+	cli->model = model;
+	return 0;
+}
+
+const struct bt_mesh_model_cb _gen_onoff_cli_cb = {
+	.init = gen_onoff_cli_init,
+};
+
+int gen_onoff_cli_status_get(struct gen_onoff_cli *cli,
+			     struct bt_mesh_msg_ctx *ctx,
+			     struct gen_onoff_status *rsp)
+{
+	NET_BUF_SIMPLE_DEFINE(msg, 2 + 0 + 4);
+	bt_mesh_model_msg_init(&msg, GEN_ONOFF_OP_GET);
+
+	return model_ackd_send(cli->model, ctx, &msg,
+			       rsp ? &cli->ack_ctx : NULL, GEN_ONOFF_OP_STATUS,
+			       rsp);
+}
+
+int gen_onoff_cli_set(struct gen_onoff_cli *cli, struct bt_mesh_msg_ctx *ctx,
```

Comment:
This requires some consideration, I think. Currently, the models only send once, acked message or not. I think it can safely be added a bit later without breaking any APIs.